atom/common/api/features.cc

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2018 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "atom/common/node_includes.h"
+#include "native_mate/dictionary.h"
+
+namespace {
+
+bool IsPDFViewerEnabled() {
+#if defined(ENABLE_PDF_VIEWER)
```

Comment:
Why don't we use `IsPDFViewerEnabled` everywhere instead of having `#ifdef`s everywhere? It'll be the same thing finally, but with only one `#ifdef` here, easier IMO.

Reply:
We shouldn't be accessing api class outside its scope. The class is just a helper for js land.

Reply:
> Why don't we use IsPDFViewerEnabled everywhere instead of having #ifdefs everywhere?

Build time checks are better than runtime-checks for anything disabled in a build time.

Reply:
Yeah, sorry! I meant something similar. This is file scope anyway, so this wont work.
I just feel that having a function call instead of `#ifdef` everywhere should be considered.
The function call will probably be optimized out, but the code will be compiled always. That wont happen for `#ifdef`.

Reply:
> but the code will be compiled always

Why do we have to compile code that we don't need? )